### DISCUSSION TITLE
Not a valid data file?

### DISCUSSION THREAD

**Discussion by benjamin-cash** [dtcenter/METplus#1080] [File I/O] [answered]
I tried a simple test using plot_data_plane to look at netcdf output from the UFS on a gaussian grid and got back the following:

plot_data_plane sfcf840.nc test.ps 'name="prateb_ave"; level="(*,*,*)";' -v 4
DEBUG 1: Opening data file: sfcf840.nc
DEBUG 4: Met2dDataFileFactory::new_met_2d_data_file() -> created new Met2dDataFile object of type "FileType_None".
ERROR  : 
ERROR  : plot_data_plane -> file "sfcf840.nc" not a valid data file
ERROR  : 

Is this a problem with how I have set up the plot_data_plane command, or is there something in the ufs gaussian grid that is causing an issue?

**Comment by georgemccabe** [comment, marked as answer]
Hi @benjamin-cash,

It looks like it is trying to interpret the file as a NetCDF file that is generated by MET, which it is not. You could try running it like this:

plot_data_plane sfcf840.nc test.ps 'name="prateb_ave"; level="(*,*)"; file_type=NETCDF_NCCF;' -v 4

This tells the app that it is a CF compliant NetCDF file. Navigate to https://met.readthedocs.io/en/latest/Users_Guide/config_options.html?highlight=NETCDF_NCCF#settings-common-to-multiple-tools and search for NETCDF_NCCF for more info.

If that does not work, then the file may not be a CF compliant file. If that is the case, you can either modify the file to include the necessary information to be read by MET or you can write a Python Embedding script to read the data and set the appropriate attributes to pass the data into MET.

If you continue to have issues, you can send the file in question and I can take a closer look.

**Reply by benjamin-cash** [reply]
That did it, thanks!

**Reply by georgemccabe** [reply]
I will also mention that the MET tools can only read a 2D grid at a time, so if the field has 3 dimensions, i.e. (\*,\*,\*), then you will have to specific a value for the 3rd dimension. For example, if the 1st dimension is time you can set level="(0,\*,\*)" to access the first time in the field. MET also supports specifying the time information in place of the index for the time dimension, so you can set level="(YYYYMMDD_HH,\*,\*)" where YYYYMMDD_HH is the year, month, day, and hour of the time you want to use, i.e. 20210811_12.
